### ISSUE TITLE
RVM and Ocra ISSUE (@ in the middle of a string)

### ISSUE THREAD

**Issue by RafaAguilar** [larsch/ocra#96]
Hello my friends,

I have an issue here, i use RVM in my RHEL server, if i run:

`ocra test.rb`

it throws:

```
/home/user/.rvm/gems/ruby-2.1.0/gems/ocra-1.3.5/bin/ocra:42:in `relative_path_from': undefined method `path' for "/home/user/.rvm/gems/ruby-2.1.0@global":String (NoMethodError)
        from /home/openproject/.rvm/gems/ruby-2.1.0/gems/ocra-1.3.5/bin/ocra:780:in `block in build_exe'
        from /home/user/.rvm/gems/ruby-2.1.0/gems/ocra-1.3.5/bin/ocra:763:in `each'
        from /home/user/.rvm/gems/ruby-2.1.0/gems/ocra-1.3.5/bin/ocra:763:in `build_exe'
        from /home/user/.rvm/gems/ruby-2.1.0/gems/ocra-1.3.5/bin/ocra:1178:in `block in <top (required)>'
```
 I debug a litle bit and i found that with other paths was no problems, here the String with the issue:

`/home/user/.rvm/gems/ruby-2.1.0@global/gems/executable-hooks-1.3.2/lib/executable-hooks/hooks.rb`

My test.rb is this:

```ruby
open('myfile.out', 'w') { |f| f.puts "Hello, world." }
```

**Comment by AvaelKross** [comment]
+

**Comment by lucasocon** [comment]
+1

**Comment by kriskova** [comment]
Same here:

```
/home/user/.rvm/gems/ruby-2.1.5/gems/ocra-1.3.5/bin/ocra:41:in `relative_path_from': undefined method `path' for "/home/user/.rvm/gems/ruby-2.1.5@global":String (NoMethodError)
	from /home/user/.rvm/gems/ruby-2.1.5/gems/ocra-1.3.5/bin/ocra:779:in `block in build_exe'
	from /home/user/.rvm/gems/ruby-2.1.5/gems/ocra-1.3.5/bin/ocra:762:in `each'
	from /home/user/.rvm/gems/ruby-2.1.5/gems/ocra-1.3.5/bin/ocra:762:in `build_exe'
	from /home/user/.rvm/gems/ruby-2.1.5/gems/ocra-1.3.5/bin/ocra:1177:in `block in <top (required)>'
```

The `feature` its got stuck: `"executable-hooks/hooks.rb"`

`path` is: `#<Ocra::Pathname:0x00000003d3d8f0 @path="/home/user/.rvm/gems/ruby-2.1.5@global/gems/executable-hooks-1.3.2/lib">`

`Host.exec_prefix` is: ` #<Ocra::Pathname:0x00000003d25e58 @path="/home/user/.rvm/rubies/ruby-2.1.5">
`

`Gem.path` is: `["/home/user/.rvm/gems/ruby-2.1.5", "/home/user/.rvm/gems/ruby-2.1.5@global"]
`

from `Gem.path`, `gemhome` is: `"/home/user/.rvm/gems/ruby-2.1.5@global"`

`relative_path_from` expected `gemhome` to be an Ocra::Pathname so it tried:

`other = Ocra::Pathname.new(other) if other.class != Ocra::Pathname`

but then i got:

```
/home/user/.rvm/rubies/ruby-2.1.5/lib/ruby/site_ruby/2.1.0/rubygems/core_ext/kernel_require.rb:54:in `require': cannot load such file -- Win32API (LoadError)
  from /home/user/.rvm/rubies/ruby-2.1.5/lib/ruby/site_ruby/2.1.0/rubygems/core_ext/kernel_require.rb:54:in `require'
  from /home/user/.rvm/gems/ruby-2.1.5/gems/ocra-1.3.5/bin/ocra:955:in `loaded_dlls'
  from /home/user/.rvm/gems/ruby-2.1.5/gems/ocra-1.3.5/bin/ocra:981:in `detect_dlls'
  from /home/user/.rvm/gems/ruby-2.1.5/gems/ocra-1.3.5/bin/ocra:846:in `build_exe'
  from /home/user/.rvm/gems/ruby-2.1.5/gems/ocra-1.3.5/bin/ocra:1180:in `block in <top (required)>'
```

**Comment by ajnye** [comment]
+

**Comment by emre801** [comment]
+1

**Comment by jamesr2323** [comment]
+

**Comment by Torq07** [comment]
+

**Comment by irahey** [comment]
+1

**Comment by Hostgame** [comment]
+

**Comment by ghuestis** [comment]
+

**Comment by 1v** [comment]
+1
```
/home/ruby215/.rvm/gems/ruby-2.1.5/gems/ocra-1.3.5/bin/ocra:41:in `relative_path_from': undefined method `path' for "/home/ruby215/.rvm/gems/ruby-2.1.5@global":String (NoMethodError)
	from /home/ruby215/.rvm/gems/ruby-2.1.5/gems/ocra-1.3.5/bin/ocra:779:in `block in build_exe'
	from /home/ruby215/.rvm/gems/ruby-2.1.5/gems/ocra-1.3.5/bin/ocra:762:in `each'
	from /home/ruby215/.rvm/gems/ruby-2.1.5/gems/ocra-1.3.5/bin/ocra:762:in `build_exe'
	from /home/ruby215/.rvm/gems/ruby-2.1.5/gems/ocra-1.3.5/bin/ocra:1177:in `block in <top (required)>'
```

**Comment by larsch** [comment]
Linux support is currently out of scope; see #98.

**Comment by larsch** [comment]
Pathname issues was fixed in fbd53dc509445313894b9b52768390bca612f2bd.
